[WIP] Update CHANGELOG for 1.14.0 release (#4112)
mlflow/mlflow: CHANGELOG.rst

Changelog
=========
1.14.0 (2021-02-18)
-------------------
MLflow 1.14.0 includes several major features and improvements:

> Support for tensor inputs using the TF serving format:

Reviewer: I think we could actually describe this in a single line item, but link to all the relevant PRs. 

```suggestion
- MLflow's model inference APIs  (`mlflow.pyfunc.predict`), built-in model serving tools (`mlflow models serve`), and model signatures now support tensor inputs. In particular, MLflow's integrations with PyTorch, TensorFlow, Keras, ONNX, and Gluon support tensor input output of the box. For more information, see https://mlflow.org/docs/latest/models.html#deploy-mlflow-models (#3894 @wentinghu ...)
```

- Add support for TF serving input to REST API (#3894, @wentinghu)
- Add support for np array-based tensors for Keras, PyTorch and TensorFlow flavors (#3808, @wentinghu)
- Add tensor input support for ONNX and gluon flavors (#4041, @tomasatdatabricks)
- Add support for tensor-based model signatures in schema enforcement (#4084, @wentinghu)

For more information, see https://mlflow.org/docs/latest/models.html#deploy-mlflow-models

More features and improvements:

> - Add SHAP Explainer logging integration (#3989, @vivekchettiar)

Reviewer: ```suggestion
- Added new `mlflow.shap.log_explainer`, `mlflow.shap.load_explainer` APIs for logging and loading shap.Explainer instances (#3989, @vivekchettiar)
```

In general when we've added new APIs or changed the behaviors of existing APIs, it's useful to mention the impacted APIs directly, so that users can look them up in the docs/try them out :)

> - Enable skinny client package build & release (#4049, @eedeleon)

Reviewer: ```suggestion
- The MLflow Python client is now available with a reduced dependency set via the ``mlflow-skinny`` PyPI package (#4049, @eedeleon)
```

> - Update client to pass request headers created by all ``RequestHeaderProviders`` (#4042, @jimmyxu-db)

Reviewer: ```suggestion
- Added new ``RequestHeaderProvider`` plugin interface for passing custom request headers with REST API requests made by the MLflow Python client (#4042, @jimmyxu-db)
```

> - Use ``SavedModel`` format by default in ``mlflow.keras.log_model`` (#4043, @harupy)

Reviewer: ```suggestion
- ``mlflow.keras.log_model`` now saves models in ``SavedModel`` format by default instead of XYZ (#4043, @harupy)
```

Maybe reword to this? What was the old model format?

> - Add support for MLeap in R (#3819, @yitao-li)

Reviewer: Similar to the comment above, could we state the specific new API that was added? e.g. "`mlflow_log_model` now supports logging MLeap models in R"?

> - Add ``log_state_dict`` / ``load_state_dict`` for logging / loading PyTorch state dict (#3705, @shrinath-suresh)

Reviewer: ```suggestion
- Added ``mlflow.pytorch.log_state_dict``, ``mlflow.pytorch.load_state_dict`` for logging / loading PyTorch state dicts (#3705, @shrinath-suresh)
```

> - Implement ``delete_artifacts`` for ``S3ArtifactRepository``  (#3958, @sklingel)

Reviewer: What does this allow users to do? Maybe
```suggestion
- `mlflow gc` can now garbage-collect artifacts stored in S3 (#3958, @sklingel)
```
